@nativescript/camera - Failed to build plugin @nativescript/imagepicker :

Build fails after adding plugin with nativescript angular, there seems to be a problem in @nativescript/imagepicker.

The console shows this issue

_What went wrong:
Execution failed for task ':verifyReleaseResources'._
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > 1 exception was raised by workers:
     com.android.builder.internal.aapt.v2.Aapt2Exception: Android resource linking failed
     C:\DEV\eSavjetnik\agrivi.services.iot.mobile\platforms\tempPlugin\imagepicker\src\main\AndroidManifest.xml:4:3-61: AAPT: error: attribute android:requestLegacyExternalStorage not found.

_BUILD FAILED in 3s
Failed to build plugin @nativescript/imagepicker : 
Error: Command gradlew.bat failed with exit code 1_ 

I also added  `android:requestLegacyExternalStorage="true"` to the android manifest in the <application> tag but the issue persists, 

EDIT: Removing the plugin and leaving `android:requestLegacyExternalStorage="true"` doesn't make the issue go away, I had to remove `android:requestLegacyExternalStorage="true"`  from manifest. I read something about it being removed as a flag in sdkVersion 30 (Android 11). Also, there is this issue which could be related to this, 
[https://github.com/NativeScript/nativescript-camera/issues/252](https://github.com/NativeScript/nativescript-camera/issues/252)

--- a/packages/imagepicker/index.android.ts
+++ b/packages/imagepicker/index.android.ts
@@ -194,7 +194,6 @@ export class ImagePicker {
 					if (resultCode === android.app.Activity.RESULT_OK) {
 						try {
 							let results = [];
-
 							let clip = data.getClipData();
 							if (clip) {
 								let count = clip.getItemCount();
@@ -203,14 +202,14 @@ export class ImagePicker {
 									if (clipItem) {
 										let uri = clipItem.getUri();
 										if (uri) {
-											let selectedAsset = new ImageAsset(UriHelper._calculateFileUri(uri));
+											let selectedAsset = new ImageAsset(uri.toString());
 											results.push(selectedAsset);
 										}
 									}
 								}
 							} else {
-								let uri = data.getData();
-								let selectedAsset = new ImageAsset(UriHelper._calculateFileUri(uri));
+								const uri = data.getData();
+								const selectedAsset = new ImageAsset(uri.toString());
 								results.push(selectedAsset);
 							}
 

--- a/packages/imagepicker/index.android.ts
+++ b/packages/imagepicker/index.android.ts
@@ -195,6 +195,7 @@ export class ImagePicker {
 						try {
 							let results = [];
 							let clip = data.getClipData();
+							const useHelper = (<any>android).os.Build.VERSION.SDK_INT <= 28;
 							if (clip) {
 								let count = clip.getItemCount();
 								for (let i = 0; i < count; i++) {
@@ -202,14 +203,16 @@ export class ImagePicker {
 									if (clipItem) {
 										let uri = clipItem.getUri();
 										if (uri) {
-											let selectedAsset = new ImageAsset(uri.toString());
+											const val = useHelper ? UriHelper._calculateFileUri(uri) : uri.toString();
+											const selectedAsset = new ImageAsset(val);
 											results.push(selectedAsset);
 										}
 									}
 								}
 							} else {
 								const uri = data.getData();
-								const selectedAsset = new ImageAsset(uri.toString());
+								const val = useHelper ? UriHelper._calculateFileUri(uri) : uri.toString();
+								const selectedAsset = new ImageAsset(val);
 								results.push(selectedAsset);
 							}
 

--- a/packages/imagepicker/platforms/android/AndroidManifest.xml
+++ b/packages/imagepicker/platforms/android/AndroidManifest.xml
@@ -1,10 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
-
-	<!--Describe the permissions, features or other configurations required by your plugin for Android. 
-		To read more about android permissions go to https://developer.android.com/guide/topics/permissions/index.html -->
-	<!--EXAMPLE: uses-permission android:name="android.permission.INTERNET"/> -->
-
 	<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
-	<application android:requestLegacyExternalStorage="true" />
 </manifest>

